firedrake/preconditioners/pmg.py

```diff
@@ -262,6 +262,35 @@ def prolongation_transfer_kernel_aij(Pk, P1):
     return kernel
 
 
+def tensor_product_space_query(V):
+    # Check if a function space V is a tensor product of
+    # either CG(N) or DG(N) on quads or hexes (same N along every direction).
+    # Additionally, return the degree N and a list of strings with the family.
+    ele = V.ufl_element()
+    cell = ele.cell()
+    isquad = cell.cellname().find("quadrilateral") >= 0  # hex == "quadrilateral * interval"
+
+    if isinstance(ele, (firedrake.VectorElement, firedrake.TensorElement)):
+        subel = ele.sub_elements()
+        ele = subel[0]
+
+    N = ele.degree()
+    issupported = True
+    if(type(N) == tuple):
+        issupported = len(set(N)) == 1
+        N = N[0]
```

Comment:
Python style is to "beg for forgiveness rather than ask for permission":

```python
N = ele.degree()
issupported = True
try:
   N, = set(N)   
except ValueError:
  # Tuple with different extents
  issupported = False
except TypeError:
  # Just a single int
  pass
```

Reply:
This will still work if we suddenly change the type of iterable used for degrees in tensor-product cells from tuples to something else.